File: ICSharpCode.Decompiler.Tests/TestCases/ILPretty/Issue1325.vb
Imports System
Imports System.IO

Module Program
    Sub Main(args As String())

    End Sub

    Sub TestCode(ByVal t As Test, ByVal i As Integer)
        Dim s As String = ""
        s += File.ReadAllText("Test.txt")
        s += "asdf"
        t.Parameterized(i) = s
        t.Unparameterized = s + "asdf"
    End Sub
End Module

Class Test
    Public Property Parameterized(ByVal i As Integer) As String
        Get
            Throw New NotImplementedException()
        End Get
        Set(value As String)
            Throw New NotImplementedException()
        End Set
    End Property

    Public Property Unparameterized As String
End Class

File: ICSharpCode.Decompiler.Tests/TestCases/ILPretty/Issue1325.cs
using System;
using System.ComponentModel;
using System.Diagnostics;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;

using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

[assembly: Embedded]
[assembly: AssemblyInformationalVersion("1.0.0")]
[assembly: AssemblyConfiguration("Debug")]
[assembly: AssemblyFileVersion("1.0.0.0")]

namespace Issue1325
{
	[StandardModule]
	internal sealed class Program
	{
		[STAThread]
		public static void Main(string[] args)
		{
		}
		public static void TestCode(Test t, int i)
		{
			string str = "";
			str += File.ReadAllText("Test.txt");
			str += "asdf";
			t.set_Parameterized(i, str);
			t.Unparameterized = str + "asdf";
		}
	}

	internal class Test
	{
		[DebuggerBrowsable(DebuggerBrowsableState.Never)]
		[CompilerGenerated]
		private string _Unparameterized;
		public string Parameterized {
			get {
				throw new NotImplementedException();
			}
			set {
				throw new NotImplementedException();
			}
		}
		public string Unparameterized {
			get;
			set;
		}
	}
}
namespace Microsoft.VisualBasic
{
	[AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Module | AttributeTargets.Class, Inherited = false)]
	[CompilerGenerated]
	[EditorBrowsable(EditorBrowsableState.Never)]
	[Embedded]
	internal sealed class Embedded : Attribute
	{
	}
}
namespace Microsoft.VisualBasic.CompilerServices
{
	[EditorBrowsable(EditorBrowsableState.Never)]
	[AttributeUsage(AttributeTargets.Class, Inherited = false)]
	[CompilerGenerated]
	[Embedded]
	internal sealed class StandardModuleAttribute : Attribute
	{
	}
}
